### PR TITLE
[HttpClient] Fix exception triggered with AsyncResponse

### PR DESCRIPTION
**Pull request by jderusse**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | https://github.com/symfony/symfony/pull/38289#issuecomment-702573975
| License       | MIT
| Doc PR        | /

When a request is replayed with the AsyncResponse and the 2nd request failed, users get an exception even when they pass `$throw=false`.

In fact, there are 2 exceptions:

1. Cannot get the content of the response twice: buffering is disabled.

https://github.com/symfony/symfony/blob/73ca97c97ab4c4fc9a7bca5fb4339fca22e0cfa1/src/Symfony/Component/HttpClient/Response/CommonResponseTrait.php#L51-L68

Because CommonResponseTrait::$content is null and `self::stream` yield only the LastChunk. The content is stays null

2. HTTP/2 429  returned for "https://httpbin.org/status/429"

https://github.com/symfony/symfony/blob/73ca97c97ab4c4fc9a7bca5fb4339fca22e0cfa1/src/Symfony/Component/HttpClient/Response/AsyncResponse.php#L209-L225

Because on the second request passthru is null, it didn't disable the the exception thrown on destruct for the wrapped response. 
This

### CODE REVIEW

File: src/Symfony/Component/HttpClient/Response/CommonResponseTrait.php [hunk 1]
```diff
@@ -54,6 +54,8 @@ public function getContent(bool $throw = true): string
             foreach (self::stream([$this]) as $chunk) {
                 if (!$chunk->isLast()) {
                     $content .= $chunk->getContent();
+                } elseif (null === $content) {
+                    $content = '';
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpClient/Response/CommonResponseTrait.php:58]: this cannot be correct (and the CI agrees :) )

Comment (jderusse) [reply src/Symfony/Component/HttpClient/Response/CommonResponseTrait.php:58]: hmm, ok, I understand... the issue in not in AsyncResponse, but in no-buffer response. Here is a reproduct case:


```php
$client = HttpClient::create();
$resp = $client->request('GET', 'https://httpbin.org/status/429', ['buffer' => false]);
$resp->getContent(false);
```

Issue is, in that case, we have 2 chunks: `FirstChunk` and `LastChunk` (without data).
`FirstChunk` is swallowed by `CommonResponseTrait::initialize` letting only the `LastChunk` that did not let time to `$content` to get data.

I need helps to fix it here.

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpClient/Response/CommonResponseTrait.php:58]: See https://github.com/symfony/symfony/pull/38388

**Comment by nicolas-grekas** [comment]
Thank you @jderusse.

**Pull request merged**